test/fixtures/ejson-cloud/service-account.yml

```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
```

Comment:
I don't think this new service account, and by extension your image pull secret, is ever being used, since your test uses the `hello-cloud` fixture set, not the `ejson-cloud` set. If you switch your test to use ejson-cloud, one option would be to make ejson-cloud's web.yaml _always_ mount this new service account and have your test assert that it does so.

Reply:
Added another test "test_pod_get_image_pull_secret_from_service_account" in ejson-cloud to verify that if a pod does not contain any ImagePullSecrets, ImagePullSecrets of the pod's ServiceAccount should be added to the pod.
Test "test_service_account_predeployed_before_unmanaged_pod" in hello_cloud is to verify that service accounts are deployed before unmanaged pods.

Reply:
Isn't that basically just testing a kubernetes feature again? That's not something the gem is doing itself.

Reply:
Following our discuss, I have removed the test for image pull secret.